src/rpm/signature/pgp.rs

```diff
@@ -66,6 +66,11 @@ where
             .push(Subpacket::critical(SubpacketData::Issuer(
                 self.secret_key.key_id(),
             )));
+        sig_cfg
+            .hashed_subpackets
+            .push(Subpacket::regular(SubpacketData::Issuer(
+                self.secret_key.key_id(),
+            )));
```

Comment:
Ideally we want to match what RPM is doing as closely as possible.  If standard RPM doesn't mark these subpackets critical and doesn't duplicate them (which based on the outputs you posted on the issue seems to be the case), then we probably shouldn't be doing that either.  

Could you try having just one "regular" copy of both the issuer and signature creation time?

Also, I think the issuer fingerprint can be handled with this:

```
     sig_cfg
            .hashed_subpackets
            .push(Subpacket::regular(SubpacketData::IssuerFingerprint(
                self.secret_key.fingerprint(),
            )));
```

Reply:
I've added the fingerprint as well, initially I wanted to do the minimum change to get signing to work.